### PR TITLE
feat: add IntegrationOptionsPlugin

### PR DESCRIPTION
**Pull request by vgupta98**
## Description
This PR introduces the `IntegrationOptionsPlugin` - a new event processing plugin that provides fine-grained control over event routing to destinations based on integration options. The plugin evaluates boolean flags in event integration settings to determine whether events should be passed to specific destinations or dropped entirely.

This plugin is for the standard integrations maintained by RudderStack.

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactor/optimization

## Implementation Details
- **New Plugin Class**: Added `IntegrationOptionsPlugin.swift` in `Sources/RudderStackAnalytics/Plugins/DeviceMode/EventProcessing/`
  - Implements priority-based logic for event filtering: destination-specific flags → "All" destinations flag → default allow
  - Handles non-boolean values gracefully by ignoring them and applying fallback logic
  - Includes comprehensive logging for dropped events
- **AnyCodable Extension**: Extended `AnyCodable` in `Extensions.swift` with `asBool()` method for safe boolean extraction from integration settings
- **Integration into DeviceMode**: Modified `IntegrationsController.swift` to automatically apply `IntegrationOptionsPlugin` to all integrations through `applyDefaultPlugins()` method
- **Project Structure**: Created new `EventProcessing` folder under `DeviceMode` plugins for better organization
- **Comprehensive Testing**: Added `IntegrationOptionsPluginTests.swift` with 9 test cases covering all scenarios:
  - All destinations enabled/disabled
  - Destination-specific enabled/disabled flags
  - Priority handling between destination-specific and "All" flags
  - Edge cases with non-boolean values, empty/nil integrations

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation (if appropriate).
- [x] I have ensured that my code follows the project's code style.
- [x] I have checked for potential performance impacts and optimized if necessary.
- [x] I have checked the code for security issues.
- [ ] I have updated the changelog (if required).

## How to test?
1. **Build and Run Tests**: Execute the test suite to verify all 9 test cases pass
   ```bash
   swift test --filter IntegrationOptionsPluginTests
   ```
2. **Integration Testing**: 
   - Send events with different integration options in your app
   - Verify events are properly filtered based on destination keys
   - Test with `"All": true/false` and destination-specific boolean flags
3. **Edge Case Testing**:
   - Send events with non-boolean integration values (strings, objects)
   - Verify events are not dropped when integration options contain complex data
   - Test with nil or empty integration options

## Breaking Changes
None. This is a purely additive feature that automatically applies to all integrations without requiring any changes to existing code or APIs.

## Maintainers Checklist
- [ ] The code has been reviewed.
- [ ] CI tests have passed.
- [ ] All necessary documentation has been updated.

## Screenshots (if applicable)
N/A - This is a backend event processing feature with no UI components.

## Additional Context
NA

## Summary by CodeRabbit

* **New Features**
  * Added event flow control based on destination-specific integration settings, allowing events to be filtered or routed based on configuration flags for individual destinations or all destinations.

* **Tests**
  * Added comprehensive test coverage for event filtering across various integration scenarios and edge cases.

### CODE REVIEW

File: Sources/RudderStackAnalytics/Helpers/Extensions.swift [hunk 1]
```diff
@@ -303,6 +303,27 @@ private func sanitizeNSNumber(_ number: NSNumber) -> Any {
...
+     *
+     * # Usage
+     * ```swift
+     * let integrations: [String: AnyCodable] = ["Firebase": AnyCodable(true)]
+     * let isEnabled = integrations["Firebase"]?.asBool() // Returns true
+     * ```
+     */
+    func asBool() -> Bool? {
+        return value as? Bool
+    }
```
Comment (SKannaniOS) [inline Sources/RudderStackAnalytics/Helpers/Extensions.swift]: This function still returns an optional Boolean. Use optional unwrapping directly instead of calling this function. Please remove it.

File: Sources/RudderStackAnalytics/Plugins/DeviceMode/EventProcessing/IntegrationOptionsPlugin.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    func intercept(event: Event) -> Event? {
+        guard let integrations = event.integrations else {
+            // No integrations specified - allow event by default
+            return event
+        }
+        
+        // First priority: Check for destination-specific boolean flag
+        if let destinationValue = integrations[destinationKey]?.asBool() {
```
Comment (SKannaniOS) [inline Sources/RudderStackAnalytics/Plugins/DeviceMode/EventProcessing/IntegrationOptionsPlugin.swift]: Instead of below
`if let destinationValue = integrations[destinationKey]?.asBool()`
simply use 
`if let destinationValue = integrations[destinationKey]?.value as? Bool`

File: Sources/RudderStackAnalytics/Plugins/DeviceMode/EventProcessing/IntegrationOptionsPlugin.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            return destinationValue ? event : {
+                logDroppedEvent(event)
+                return nil
+            }()
+        }
+        
+        // Second priority: Check for "All" destinations boolean flag
+        if let allValue = integrations["All"]?.asBool() {
```
Comment (SKannaniOS) [inline Sources/RudderStackAnalytics/Plugins/DeviceMode/EventProcessing/IntegrationOptionsPlugin.swift]: Instead of below
`if let allValue = integrations["All"]?.asBool()`
simply use 
`if let allValue = integrations["All"]?.value as? Bool`

**Review by SKannaniOS — APPROVED** [review]
Looks good
